go/src/koding/klientctl/endpoint/machine/mount.go

```diff
@@ -201,25 +206,80 @@ func (c *Client) Umount(options *UmountOptions) (err error) {
 		return errors.New("invalid nil options")
 	}
 
-	// TODO: ask user to confirm unmounting.
-	fmt.Fprintf(os.Stdout, "Unmounting %s...\n", options.Identifier)
+	// Get all mounts if all option is set.
+	identifiers := options.Identifiers
+	if options.All {
+		mids, err := c.allMounts()
+		if err != nil {
+			return err
+		}
+		identifiers = mids
+	}
 
-	// Remove mount.
-	umountReq := &machinegroup.UmountRequest{
-		Identifier: options.Identifier,
+	// If there are no mounts we return nil.
+	if len(identifiers) == 0 {
+		fmt.Fprintf(os.Stdout, "There is nothing to unmount\n")
+		return nil
```

Comment:
It is backward-compatibile, `len(options.Identifiers) == 1` is when user provides: "kd umount <mount>". The case above (`len(options.Identifiers) == 0`) covers the command "kd umount -a" when there is no mounts. Eg.  "kd umount --force" will result in error [here](https://github.com/koding/koding/pull/11096/files#diff-6c91a42efe84e0738dea5008055828c2R151).